src/main/java/com/dao/LocalJsonMailDAO.java

```diff
@@ -0,0 +1,97 @@
+package com.dao;
+
+import com.dto.MailDto;
+import com.model.Mail;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.springframework.stereotype.Repository;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Repository
+public class LocalJsonMailDAO implements MailDao {
+
+    private List<MailDto> data = new ArrayList<>();
+
+    @PostConstruct
+    public void init() {
+        load();
+    }
+
+    @PreDestroy
+    public void destroy() {
+        save();
+    }
+
+    @Override
+    public void add(MailDto mailDto) {
+        data.add(mailDto);
+    }
+
+    @Override
+    public List<Mail> getAll() {
+        return data.stream()
+                .map(maildto -> new Mail(data.indexOf(maildto), maildto))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public void update(int id, String date) {
+        if (id > data.size()) return;
```

Comment:
In this case your client will not know when an update fails due to non-existent entity. You should throw an exception and return an appropriate status to the client